File: src/Python/701-800/762.CountPrimeSetBits.py

class Solution:
    def countPrimeSetBits(self, L, R):
        """
        :type L: int
        :type R: int
        :rtype: int
        """
        primes = {2, 3, 5, 7, 11, 13, 17, 19}
        count = 0
        for i in range(L, R + 1):
            if bin(i).count('1') in primes:
                count += 1
        return count

if __name__ == '__main__':
    solution = Solution()
    print(solution.countPrimeSetBits(6, 10))
    print(solution.countPrimeSetBits(10, 15))
else:
    pass

File: src/Java/701-800/762.CountPrimeSetBits.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public int countPrimeSetBits(int L, int R) {
        int[] _primes = new int[]{0, 0, 1, 1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 0};
        int count = 0;
        for (int i = L; i <= R; i++) {
            int bits = 0;
            for (int n = i; n > 0; n >>= 1)
                bits += n & 1;
            if (_primes[bits] == 1)
                count += 1;
        }
        return count;
    }
}